flink-clients/src/main/java/org/apache/flink/client/program/OptimizerPlanEnvironment.java

```diff
@@ -74,9 +76,11 @@ public FlinkPlan getOptimizedPlan(PackagedProgram prog) throws ProgramInvocation
 		PrintStream originalOut = System.out;
 		PrintStream originalErr = System.err;
 		ByteArrayOutputStream baos = new ByteArrayOutputStream();
-		System.setOut(new PrintStream(baos));
+		TeeOutputStream combinedStdOut = new TeeOutputStream(originalOut, baos);
+		System.setOut(new PrintStream(combinedStdOut));
 		ByteArrayOutputStream baes = new ByteArrayOutputStream();
-		System.setErr(new PrintStream(baes));
+		TeeOutputStream combinedStdErr = new TeeOutputStream(originalErr, baes);
```

Comment:
> Do we need to close these streams? Additionally, its not clear to me why we need to trap the output. It seems to only be used for the exception at line 109. Couldn't the client just refer to the logs instead?

Thank you for your comments. 
It depends on the underlying streams to decide whether we need to close the streams. The underlying steams are stdout/stderr, and byte array streams. So it is OK not to close them. However, for the sake of completeness, it is better to close them.
The purpose of the trapping is that the output will be both encoded in the exception, and printed in the console, so that the user can easily see it.

Reply:
Thanks. I guess it seems inconsistent to send the stdout/stderr information along with the error in only one out of the three exceptions handled by this method. Admittedly, I don't understand the codebase well enough to know why this has been written this way. Approving as your changes address FLINK-12259